### ISSUE TITLE
Multiple lists of auth keys in Settings notecard overwrite 

### ISSUE THREAD

**Issue by Medea-Destiny** [OpenCollarTeam/OpenCollar#178]
Reported by Himeko (himechan) in world:

> I found an issue with the settings reader. Due to limitations on how long a notecard line can be, it can't read more than 6 or so keys for putting into owners, blocked and trusted. Sure, you can split those 3 up on different lines, but you couldn't add more than that to any one of them. Putting for example owner on 2 lines overwrites the first one with the second instead of adding them together, which does make sense.

Thanks for reporting, Himeko!  With the extended scope for owner+trusted+blacklist it does make sense to look at this. Trusted and Blacklist are both options that it's easy to imagine people wanting to exceed the 255 character limit for. 6 looks right as the limit for me.

Settings currently replace a list entry by token. That really makes sense for most things, so we'd need to have a special handling for owner/trust/blacklist tokens to append rather than replace the list element.

**Comment by MikuGit** [comment]
Thank you for posting it for me~
One way to solve it, no idea if the best or not, would be to add a new tag for each of the 3 categories. "addowner", "addblacklist" and "addtrusted" that work the same way as the other, but adding to the lists instead of overwriting them.

Another alternative could be to have an end character, like ";" when a row actually ends and make you allowed to line break it indefinitely.

Neither alternative is really making it easy for the end user though as both are unintuitive as SL allows you to write long rows while the script only reads 255.

Perhaps making such settings always be 1 entry per row between a start and an end tag?

To avoid memory leaks by not clearing lists though, there'd need to be some distinction between adding to a list and overwriting it. Else you might fill auth up with a bunch of people that shouldn't be there.

**Comment by Medea-Destiny** [comment]
Himeko has also pointed out that the settings script still uses the fixed number limits per auth type rather than the 28 keys in total limit.
linest 240-242
```
integer iTest = llGetListLength(lOut);
if (sToken == "owner" &&  iTest == 3)  jump next;
else if (sToken == "trust" &&  iTest == 15)  jump next;
else if (sToken == "block" &&  iTest == 9)  jump next;
```
To fix this, we'll have to extract the current lists for the other two tokens from g_lSettings, add the length of those to iTest and jump next; at 28 (the current combined hardlimit).

**Comment by zontreck** [comment]
Or.. We move to save by either uuid or name. I already added in name 2 key checking into the Auth script

**Comment by zontreck** [comment]
By saving by name in the notecard we have potentially enough room

**Comment by zontreck** [comment]
Never said it was the only solution... rarely will we have a name that is longer than a UUID though. I do think we should have AUTH_trust, block, and owner on separate lines though as a start. And as for owner, trust and block lines that are too long, i think the simpler option isnt a character to break the lines, but rather we could just make it detect if we havent put a token on the next line and then continue the last token's data.

**Comment by MikuGit** [comment]
Not putting a token could very well also work, indeed. Though it has to be conveyed to the user that they can only put so and so many entries on each line. Most intuitive would perhaps be to just allow one per line. Otherwise some warning message telling them about it when it happens.

**Comment by zontreck** [comment]
This issue is resolved, see my resolution on #234 for more information.
